spec/support/protos/optional_v3_fields.pb.rb

```diff
@@ -0,0 +1,22 @@
+# encoding: utf-8
+
+##
+# This file is auto-generated. DO NOT EDIT!
+#
+require 'protobuf'
+
+
+##
+# Message Classes
+#
+class SomethingWithOptionalFields < ::Protobuf::Message; end
+
+
+##
+# Message Fields
+#
+class SomethingWithOptionalFields
+  optional :string, :i_am_optional, 1
+  optional :string, :i_am_not_optional, 2
```

Comment:
Should this omit the `optional` flag?

Reply:
Ruby-protobuf is v2 where all fields are optional or required. The DSL depends on one of these being specified. The "optional v3" aspect doesn't really translate to v2. So long as they're wire compatible, we should be OK. Asserting that they are wire compatible is important before merging this PR.

Reply:
👍